File: app/src/main/java/com/diu/yk_games/line2box/DisplayFragment.java
package com.diu.yk_games.line2box;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.ColorStateList;
import android.graphics.PorterDuff;
import android.graphics.drawable.ColorDrawable;
import android.media.MediaPlayer;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;
import java.util.ArrayList;
import java.util.Objects;


public class DisplayFragment extends Fragment
{


    private final String TAG="Dis_frag";
    ArrayList<DataStore> dsList;
    String bestScore="\n\n\nNetwork Error";
    Context context;
    GameProfile p1Pro,p2Pro;
    SharedPreferences sharedPref;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Initialize dataset, this data would usually come from a local content provider or
        // remote server.
        context=getContext();
        sharedPref = requireContext().getSharedPreferences(
                getString(R.string.preference_file_key), Context.MODE_PRIVATE);
    }


    @SuppressLint("SetTextI18n")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState)
    {

        View v=inflater.inflate(R.layout.fragment_display, container, false);
        TextView lbs= v.findViewById(R.id.lastBestScore);
        dsList = new ArrayList<>();

        //FirebaseDatabase database = FirebaseDatabase.getInstance();
        //DatabaseReference myRef = database.getReference("ScoreBoard");

//        myRef.child("Last Best Player").addValueEventListener(new ValueEventListener() {
//            @SuppressLint("SetTextI18n")
//            @Override
//            public void onDataChange(@NonNull DataSnapshot dataSnapshot)
//            {
//                if(dataSnapshot.exists())
//                {
//                    bestScore = dataSnapshot.getValue(String.class);
//                    lbs.setText("Last Best Score is from:     "+bestScore);
//                }
//            }
//            @Override public void onCancelled(@NonNull DatabaseError error) {
//                Log.w(TAG, "Failed to read value.", error.toException());}
//        });
//        myRef.child("allScore").addChildEventListener(new ChildEventListener()
//        {
//            @Override
//            public void onChildAdded(@NonNull DataSnapshot dataSnapshot, String s)
//            {
//                if(dataSnapshot.exists())
//                {
//                    DataStore ds = dataSnapshot.getValue(DataStore.class);
//                    assert ds != null;
//                    dsList.add(0,ds);
//                    MyListAdapter adapter=new MyListAdapter(getActivity(),dsList);  //
//                    ListView list = v.findViewById(R.id.showScoreList);
//                    list.setAdapter(adapter);
//                }
//
//            }
//            @Override public void onChildChanged(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {}
//            @Override public void onChildRemoved(@NonNull DataSnapshot dataSnapshot) {}
//            @Override public void onChildMoved(@NonNull DataSnapshot dataSnapshot, @Nullable String s) {}
//            @Override public void onCancelled(@NonNull DatabaseError databaseError) {
//                Log.w(TAG, "Failed to read value.", databaseError.toException());}
//
//        });

        FirebaseFirestore db = FirebaseFirestore.getInstance();
        //Source source = Source.CACHE;
        db.collection("LastBestPlayer").document("LastBestPlayer")
                .get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
            @Override
            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                if (task.isSuccessful()) {
                    DocumentSnapshot document = task.getResult();
                    Log.d(TAG, "Cached document data: " + document.getData());
                    bestScore= Objects.requireNonNull(Objects.requireNonNull(document.getData()).get("info")).toString();
                    lbs.setText("\uD83D\uDC51 "+bestScore);
                } else {Log.d(TAG, "Cached get failed: ", task.getException());}
            }
        });
        db.collection("ScoreBoard")
                .get()
                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
                    @Override
                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
                        if (task.isSuccessful()) {
                            for (QueryDocumentSnapshot document : task.getResult())
                            {
                                DataStore ds = document.toObject(DataStore.class);
                                dsList.add(0,ds);
                            }
                            try {

                                MyListAdapter adapter=new MyListAdapter(getContext(),dsList);  //
                                ListView list = v.findViewById(R.id.showScoreList);
                                list.setAdapter(adapter);
                                list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
                                    @Override
                                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                                        DataStore gamerPro = (DataStore) list.getItemAtPosition(position);

                                        if(gamerPro.plr1Id.equals("offline"))
                                            Toast.makeText(context, "Offline match doesn't have Profile Info.", Toast.LENGTH_SHORT).show();
                                        else if(gamerPro.plr1Id.equals(""))
                                            Toast.makeText(context, "Old match doesn't have Profile Info.", Toast.LENGTH_SHORT).show();
                                        else
                                        {
                                            if(!sharedPref.getBoolean("muted", false))
                                            {
                                                MediaPlayer mediaPlayer = MediaPlayer.create(context, R.raw.btn_click_ef);
                                                mediaPlayer.start();
                                                mediaPlayer.setOnCompletionListener(MediaPlayer::release);
                                            }
                                            AlertDialog.Builder builder = new AlertDialog.Builder(context);
                                            View v = LayoutInflater.from(context).inflate(
                                                    R.layout.dialog_layout_scr_globe, parent.findViewById(R.id.scoreDetailsLayoutDialog)
                                            );
                                            builder.setView(v);

                                            Log.d(TAG, position+ "onItemClick: 1id "+gamerPro.plr1Id);
                                            Log.d(TAG, "onItemClick: 2id "+gamerPro.plr2Id);
                                            db.collection("gamerProfile").document(gamerPro.plr1Id)
                                                    .get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                                                        @SuppressLint("SetTextI18n")
                                                        @Override
                                                        public void onSuccess(DocumentSnapshot documentSnapshot) {
                                                            if(documentSnapshot.exists())
                                                            {
                                                                String[] scr=gamerPro.redData.split(" ");
                                                                Log.d(TAG, "onSuccess: scr "+scr[scr.length-1]);
                                                                ((TextView)v.findViewById(R.id.plr1Score)).setText(scr[scr.length-1]);
                                                                ((TextView)v.findViewById(R.id.plr1Cup)).setText(gamerPro.plr1Cup);
                                                                Log.d(TAG, "onSuccess: cup "+gamerPro.plr1Cup);
                                                                p1Pro = Objects.requireNonNull(documentSnapshot.toObject(GameProfile.class));
                                                                if(!p1Pro.countryEmoji.equals(""))
                                                                    ((TextView)v.findViewById(R.id.plr1Flag)).setText(p1Pro.countryEmoji);
                                                                ((TextView)v.findViewById(R.id.plr1Nm)).setText(p1Pro.nm);
                                                                Log.d(TAG, "onSuccess: nm "+ p1Pro.nm);
                                                                ((TextView)v.findViewById(R.id.plr1Lvl)).setText(""+p1Pro.lvl);

                                                            }
                                                        }
                                                    });
                                            db.collection("gamerProfile").document(gamerPro.plr2Id)
                                                    .get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                                                        @SuppressLint("SetTextI18n")
                                                        @Override
                                                        public void onSuccess(DocumentSnapshot documentSnapshot) {
                                                            if(documentSnapshot.exists())
                                                            {
                                                                String[] scr=gamerPro.blueData.split(" ");
                                                                ((TextView)v.findViewById(R.id.plr2Score)).setText(scr[scr.length-1]);
                                                                ((TextView)v.findViewById(R.id.plr2Cup)).setText(gamerPro.plr2Cup);

                                                                p2Pro = Objects.requireNonNull(documentSnapshot.toObject(GameProfile.class));
                                                                if(!p2Pro.countryEmoji.equals(""))
                                                                    ((TextView)v.findViewById(R.id.plr2Flag)).setText(p2Pro.countryEmoji);
                                                                ((TextView)v.findViewById(R.id.plr2Nm)).setText(p2Pro.nm);
                                                                ((TextView)v.findViewById(R.id.plr2Lvl)).setText(""+p2Pro.lvl);

                                                            }
                                                            final AlertDialog alertDialog = builder.create();

                                                            if (alertDialog.getWindow() != null) {
                                                                alertDialog.getWindow().setBackgroundDrawable(new ColorDrawable(0));
                                                            }
                                                            try {alertDialog.show();}
                                                            catch (Exception e) {e.printStackTrace();}
                                                        }

                                                    });


                                            v.findViewById(R.id.linLayoutPlr1).setOnClickListener(new View.OnClickListener() {
                                                @Override
                                                public void onClick(View view) {
                                                    if(!sharedPref.getBoolean("muted", false))
                                                    {
                                                        MediaPlayer mediaPlayer = MediaPlayer.create(context, R.raw.btn_click_ef);
                                                        mediaPlayer.start();
                                                        mediaPlayer.setOnCompletionListener(MediaPlayer::release);
                                                    }
                                                    AlertDialog.Builder builder = new AlertDialog.Builder(context);
                                                    View v2 = LayoutInflater.from(context).inflate(
                                                            R.layout.dialog_layout_profile, parent.findViewById(R.id.profileLayoutDialog)
                                                    );
                                                    builder.setView(v2);
                                                    LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
                                                            LinearLayout.LayoutParams.WRAP_CONTENT,
                                                            LinearLayout.LayoutParams.WRAP_CONTENT
                                                    );
                                                    params.setMargins(60, 0, 60, 0);
                                                    LinearLayout.LayoutParams params2 = new LinearLayout.LayoutParams(
                                                            LinearLayout.LayoutParams.MATCH_PARENT,
                                                            LinearLayout.LayoutParams.MATCH_PARENT
                                                    );
                                                    v2.findViewById(R.id.linearLayoutFrame).setLayoutParams(params);
                                                    v2.findViewById(R.id.linearLayoutFrame).setBackgroundTintList(ColorStateList.valueOf(ContextCompat.getColor(context, R.color.cocX)));
                                                    v2.findViewById(R.id.linearLayoutFrame).setBackgroundTintMode(PorterDuff.Mode.ADD);
                                                    if(!p1Pro.countryNm.equals(""))
                                                        ((TextView) v2.findViewById(R.id.countryTxt)).setText(p1Pro.countryNm+" "+p1Pro.countryEmoji);
                                                    else
                                                        v2.findViewById(R.id.countryLayout).setVisibility(View.GONE);
                                                    ((TextView) v2.findViewById(R.id.lvlTxt)).setText(""+p1Pro.lvl);
                                                    ((TextView) v2.findViewById(R.id.coinHave)).setText(""+p1Pro.coin);
                                                    ((TextView) v2.findViewById(R.id.matchPlayedTxt)).setText(""+p1Pro.matchPlayed);
                                                    ((TextView) v2.findViewById(R.id.matchWonTxt)).setText(""+p1Pro.matchWinMulti);
                                                    EditText nmEditText = v2.findViewById(R.id.nmTxt);
                                                    nmEditText.setEnabled(false);
                                                    nmEditText.setText(p1Pro.nm);
                                                    //nmEditText.setVisibility(View.GONE);
                                                    ((TextView) v2.findViewById(R.id.profileTitle)).setTextSize(28);

                                                    v2.findViewById(R.id.profileShapeLayout).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.nmEditBtn).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.nmLTxt).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.themeBox).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.countryLTxt).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.buttonSaveInfo).setVisibility(View.GONE);
                                                    final AlertDialog alertDialog = builder.create();

                                                    if (alertDialog.getWindow() != null) {
                                                        alertDialog.getWindow().setBackgroundDrawable(new ColorDrawable(0));
                                                    }
                                                    try {alertDialog.show();}
                                                    catch (Exception e) {e.printStackTrace();}

                                                }
                                            });
                                            v.findViewById(R.id.linLayoutPlr2).setOnClickListener(new View.OnClickListener() {
                                                @Override
                                                public void onClick(View view) {
                                                    if(!sharedPref.getBoolean("muted", false))
                                                    {
                                                        MediaPlayer mediaPlayer = MediaPlayer.create(context, R.raw.btn_click_ef);
                                                        mediaPlayer.start();
                                                        mediaPlayer.setOnCompletionListener(MediaPlayer::release);
                                                    }
                                                    AlertDialog.Builder builder = new AlertDialog.Builder(context);
                                                    View v2 = LayoutInflater.from(context).inflate(
                                                            R.layout.dialog_layout_profile, parent.findViewById(R.id.profileLayoutDialog)
                                                    );
                                                    builder.setView(v2);
                                                    LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
                                                            LinearLayout.LayoutParams.WRAP_CONTENT,
                                                            LinearLayout.LayoutParams.WRAP_CONTENT
                                                    );
                                                    params.setMargins(420, 0, 60, 0);
                                                    v2.findViewById(R.id.linearLayoutFrame).setLayoutParams(params);
                                                    v2.findViewById(R.id.linearLayoutFrame).setBackgroundTintList(ColorStateList.valueOf(ContextCompat.getColor(context, R.color.cocX)));
                                                    v2.findViewById(R.id.linearLayoutFrame).setBackgroundTintMode(PorterDuff.Mode.ADD);

                                                    if(!p2Pro.countryNm.equals(""))
                                                        ((TextView) v2.findViewById(R.id.countryTxt)).setText(p2Pro.countryNm+" "+p2Pro.countryEmoji);
                                                    else
                                                        v2.findViewById(R.id.countryLayout).setVisibility(View.GONE);
                                                    ((TextView) v2.findViewById(R.id.lvlTxt)).setText(""+p2Pro.lvl);
                                                    ((TextView) v2.findViewById(R.id.coinHave)).setText(""+p2Pro.coin);
                                                    ((TextView) v2.findViewById(R.id.matchPlayedTxt)).setText(""+p2Pro.matchPlayed);
                                                    ((TextView) v2.findViewById(R.id.matchWonTxt)).setText(""+p2Pro.matchWinMulti);
                                                    EditText nmEditText = v2.findViewById(R.id.nmTxt);
                                                    nmEditText.setEnabled(false);
                                                    nmEditText.setText(p2Pro.nm);
                                                    //nmEditText.setVisibility(View.GONE);
                                                    ((TextView) v2.findViewById(R.id.profileTitle)).setTextSize(28);

                                                    v2.findViewById(R.id.profileShapeLayout).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.nmEditBtn).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.nmLTxt).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.themeBox).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.countryLTxt).setVisibility(View.GONE);
                                                    v2.findViewById(R.id.buttonSaveInfo).setVisibility(View.GONE);
                                                    final AlertDialog alertDialog = builder.create();

                                                    if (alertDialog.getWindow() != null) {
                                                        alertDialog.getWindow().setBackgroundDrawable(new ColorDrawable(0));
                                                    }
                                                    try {alertDialog.show();}
                                                    catch (Exception e) {e.printStackTrace();}

                                                }
                                            });




                                        }
                                    }
                                });
                            }
                            catch (NullPointerException npe) {npe.printStackTrace();}
                        } else {Log.d(TAG, "Error getting documents: ", task.getException());}
                    }
                });
        return v;
    }

}

File: app/src/main/java/com/diu/yk_games/line2box/presentation/main/DisplayFragment.kt
package com.diu.yk_games.line2box.presentation.main

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.content.Context
import android.content.res.ColorStateList
import android.graphics.PorterDuff
import android.graphics.drawable.ColorDrawable
import android.media.MediaPlayer
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.diu.yk_games.line2box.R
import com.diu.yk_games.line2box.databinding.DialogLayoutProfileBinding
import com.diu.yk_games.line2box.databinding.DialogLayoutScrGlobeBinding
import com.diu.yk_games.line2box.databinding.FragmentDisplayBinding
import com.diu.yk_games.line2box.model.DataStore
import com.diu.yk_games.line2box.model.GameProfile
import com.diu.yk_games.line2box.pref
import com.diu.yk_games.line2box.presentation.ScoreListAdapter
import com.diu.yk_games.line2box.util.gone
import com.diu.yk_games.line2box.util.setBounceClickListener
import com.diu.yk_games.line2box.util.toast
import com.google.firebase.Firebase
import com.google.firebase.firestore.firestore
import com.google.firebase.firestore.toObject

class DisplayFragment : Fragment() {
    lateinit var binding: FragmentDisplayBinding
    private var dsList= mutableListOf<DataStore>()
    private var bestScore = "\n\n\nNetwork Error"
    private lateinit var p1Pro: GameProfile
    private lateinit var p2Pro: GameProfile
    private lateinit var context: Context

    private val scoreListAdapter by lazy { ScoreListAdapter() }

    companion object{
        private const val TAG = "DisplayFragment"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Initialize dataset, this data would usually come from a local content provider or
        // remote server.
        context = requireContext()
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentDisplayBinding.inflate(layoutInflater)
        return binding.root
    }

    @SuppressLint("SetTextI18n")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.showScoreList.adapter = scoreListAdapter
        val db = Firebase.firestore
        //Source source = Source.CACHE;
        db.collection("LastBestPlayer").document("LastBestPlayer")
            .get().addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    val document = task.result
                    Log.d(TAG, "Cached document data: " + document.data)
                    bestScore = document.data?.get("info")?.toString().orEmpty()
                    binding.lastBestScore.text = "\uD83D\uDC51 $bestScore"
                } else {
                    Log.d(TAG, "Cached get failed: ", task.exception)
                }
            }
        db.collection("ScoreBoard")
            .orderBy("time")
            .limitToLast(100)
            .get()
            .addOnSuccessListener { task ->
                task.documents.forEach {
                    it.toObject<DataStore>()?.let {ds->
                        dsList.add(0, ds)
                    }
                }
                Log.d(TAG, "isSuccessful: ${dsList.size}")
                scoreListAdapter.submitList(dsList)
            }
        scoreListAdapter.onClickListener = { gamerPro ->
            if ((gamerPro.plr1Id == "offline"))
                toast("Offline match doesn't have Profile Info.")
            else if ((gamerPro.plr1Id == ""))
                toast("Old match doesn't have Profile Info.")
            else {
                if (!pref.getBoolean("muted", false)) {
                    val mediaPlayer =
                        MediaPlayer.create(context, R.raw.btn_click_ef)
                    mediaPlayer.start()
                    mediaPlayer.setOnCompletionListener(MediaPlayer::release)
                }
                val builder = AlertDialog.Builder(context)
                val dialogBinding = DialogLayoutScrGlobeBinding.inflate(layoutInflater, null, false)
                builder.setView(dialogBinding.root)
                Log.d(TAG, "onItemClick: 1id " + gamerPro.plr1Id)
                Log.d(TAG, "onItemClick: 2id " + gamerPro.plr2Id)
                db.collection("gamerProfile").document(gamerPro.plr1Id)
                    .get()
                    .addOnSuccessListener { documentSnapshot ->
                        val gp = documentSnapshot.toObject<GameProfile>()
                        if (gp != null) {
                            val scr = gamerPro.redData.split(" ".toRegex())
                                .dropLastWhile { it.isEmpty() }
                                .toTypedArray()
                            Log.d(TAG, "onSuccess: scr " + scr[scr.size - 1])
                            dialogBinding.plr1Score.text = scr[scr.size - 1]
                            dialogBinding.plr1Cup.text = gamerPro.plr1Cup
                            Log.d(TAG, "onSuccess: cup " + gamerPro.plr1Cup)
                            p1Pro = gp
                            if (p1Pro.countryEmoji != "")
                                dialogBinding.plr1Flag.text = p1Pro.countryEmoji
                            dialogBinding.plr1Nm.text = p1Pro.nm
                            Log.d(TAG, "onSuccess: nm " + p1Pro.nm)
                            dialogBinding.plr1Lvl.text = "" + p1Pro.lvl
                        }
                    }
                db.collection("gamerProfile").document(gamerPro.plr2Id)
                    .get().addOnSuccessListener { documentSnapshot ->
                        val gp = documentSnapshot.toObject<GameProfile>()
                        if (gp != null) {
                            val scr = gamerPro.blueData.split(" ".toRegex())
                                .dropLastWhile { it.isEmpty() }
                                .toTypedArray()
                            dialogBinding.plr2Score.text = scr[scr.size - 1]
                            dialogBinding.plr2Cup.text = gamerPro.plr2Cup
                            p2Pro = gp
                            if (p2Pro.countryEmoji != "")
                                dialogBinding.plr2Flag.text = p2Pro.countryEmoji
                            dialogBinding.plr2Nm.text = p2Pro.nm
                            dialogBinding.plr2Lvl.text = "" + p2Pro.lvl
                        }
                        val alertDialog = builder.create()
                        alertDialog.window?.setBackgroundDrawable(ColorDrawable(0))
                        try { alertDialog.show() }
                        catch (e: Exception) { e.printStackTrace() }
                    }
                dialogBinding.linLayoutPlr1
                    .setBounceClickListener {
                        onPlayerProfileClick(p1Pro)
                    }
                dialogBinding.linLayoutPlr2
                    .setBounceClickListener {
                        onPlayerProfileClick(p2Pro, 420)
                    }
            }
        }
    }

    @SuppressLint("SetTextI18n")
    private fun onPlayerProfileClick(profile: GameProfile, marginLeft: Int = 60) {
        if (!pref.getBoolean("muted", false)) {
            val mediaPlayer = MediaPlayer.create(context, R.raw.btn_click_ef)
            mediaPlayer.start()
            mediaPlayer.setOnCompletionListener(MediaPlayer::release)
        }
        val builder2 = AlertDialog.Builder(context)
        val dBinding = DialogLayoutProfileBinding.inflate(layoutInflater, null, false)

        builder2.setView(dBinding.root)
        val params = LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.WRAP_CONTENT,
            LinearLayout.LayoutParams.WRAP_CONTENT
        )
        params.setMargins(marginLeft, 0, 60, 0)
        dBinding.linearLayoutFrame.apply {
            layoutParams = params
            backgroundTintList = ColorStateList.valueOf(ContextCompat.getColor(context, R.color.cocX))
            backgroundTintMode = PorterDuff.Mode.ADD
        }
        dBinding.apply {
            if (profile.countryNm != "")
                countryTxt.text = profile.countryNm + " " + profile.countryEmoji
            else
                countryLayout.gone()
            lvlTxt.text = "" + profile.lvl
            coinHave.text = "" + profile.coin
            matchPlayedTxt.text = "" + profile.matchPlayed
            matchWonTxt.text = "" + profile.matchWinMulti
            nmTxt.isEnabled = false
            nmTxt.setText(profile.nm)
            profileTitle.textSize = 28f
            profileShapeLayout.gone()
            nmEditBtn.gone()
            nmLTxt.gone()
            themeBox.gone()
            countryLTxt.gone()
            buttonSaveInfo.gone()
        }
        val alertDialog = builder2.create()
        alertDialog.window?.setBackgroundDrawable(ColorDrawable(0))
        dBinding.root.setOnClickListener {
            alertDialog.dismiss()
        }
        try {
            alertDialog.show()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}